orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -408,18 +408,32 @@ IMPLEMENT_GRADIENT_BUILDER(GetSplitGradient) {
 }
 
 IMPLEMENT_GRADIENT_BUILDER(GetConcatGradient) {
-  //TODO: split attribute should be used!!!
-  //AttributeProto split = MakeAttribute("split", std::vector<int64_t>());
+  auto attributes = SrcNodeAttributes();
+  ORT_ENFORCE(attributes.at("axis").has_i());
+  auto axis = attributes.at("axis").i();
 
+  std::vector<int64_t> split_attribute(GetSrcNodeInputSize());
```

Comment:
Your comment is right that this would only work the cases that concat on the static shape. 

In the case of dynamic case, the shape is symbolic, can will hit the else case in line 423

Reply:
I think Split can be the optimized implementation compared to Slice, so current code still valid and it's enough for current usage. We should add more code to use Slice op when shape is symbolic instead of going to line 423. Maybe we can do this later instead of in this PR?